File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& arr) {
        int n=arr.size();

        while(arr[0] != arr[arr[0]])
        {
            swap(arr[0], arr[arr[0]]);
        }

        return arr[0];
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, arr: List[int]) -> int:
        n = len(arr)
        temp = [0]*n
        
        for i in range(n):
            temp[arr[i]] += 1

        for i in range(0,n):
            if temp[i]>1:
                return i 

        return 0